swift/extractor/SwiftDispatcher.h

```diff
@@ -100,12 +109,12 @@ class SwiftDispatcher {
   // it actually gets emitted to handle recursive cases such as recursive calls, or recursive type
   // declarations
   template <typename E>
-  TrapLabel<ToTag<E>> assignNewLabel(E* e) {
-    assert(waitingForNewLabel == getCanonicalPointer(e) && "assignNewLabel called on wrong entity");
-    auto label = getLabel<ToTag<E>>();
+  TrapLabelOf<E> assignNewLabel(E* e) {
+    assert(waitingForNewLabel == Store::Handle{e} && "assignNewLabel called on wrong entity");
```

Comment:
Nit: Do we need to construct a `Store::Handle` or can we just compare to `e` and rely on an implicit conversion?

Reply:
no, it did not work. If I get it correctly, the compiler is not able to apply the two implicit casts (say, from `VarDecl*` to `Decl*` and from `Decl*` to `Store::Handle`), so it needs a bit of a hand